src/core/audio/main_audio.cpp

```diff
@@ -59,8 +56,8 @@ namespace otto::audio {
       util::audio::add_all(mix, mtrn);
     }
 
-
-    IF_DEBUG({
+#if OTTO_DEBUG_UI
```

Comment:
I felt that the function-macro was a little odd.

I have some thoughts on these pieces.  Firstly, it seems to me that the debug *info* should be separated from the debug *ui*.  Even in a "production" build, it may be useful to have the app continually gather 'debug' metrics.  It would make it possible to configure the device to "report" performance data to a web endpoint (if configured correctly).

However, production builds would *never* have debug *ui* enabled.

Something to consider for later down the road.

Reply:
the main thing is, i like being able to do some slightly performance hungry metrics. A lot of this is in tight loops, and there is really no reason to also fit some debug graph into cache on a production build. But yeah, separation of metrics and the ui sounds good